ui/imports/shared/controls/TransactionDelegate.qml

```diff
@@ -63,133 +118,393 @@ StatusListItem {
                                             isModelDataValid ?
                                                 Utils.compactAddress(modelData.from, 4) :
                                                 ""
-    state: "normal"
+
+    property StatusAssetSettings statusIconAsset: StatusAssetSettings {
+        width: 12
+        height: 12
+        bgWidth: width + 2
+        bgHeight: bgWidth
+        bgRadius: bgWidth / 2
+        bgColor: root.color
+        color: "transparent"
+        name: {
+            switch(root.transactionStatus) {
+            case TransactionDelegate.TransactionStatus.Pending:
+                return Style.svg("transaction/pending")
+            case TransactionDelegate.TransactionStatus.Verified:
+                return Style.svg("transaction/verified")
+            case TransactionDelegate.TransactionStatus.Finished:
+                return Style.svg("transaction/finished")
+            case TransactionDelegate.TransactionStatus.Failed:
+                return Style.svg("transaction/failed")
+            default:
+                return ""
+            }
+        }
+    }
+
+    property StatusAssetSettings tokenIconAsset: StatusAssetSettings {
+        width: 18
+        height: 18
+        bgWidth: width
+        bgHeight: height
+        bgColor: "transparent"
+        color: "transparent"
+        isImage: !loading
+        name: root.tokenImage
+        isLetterIdenticon: loading
+    }
+
+    enum TransactionType {
+        Send,
+        Receive,
+        Buy,
+        Sell,
+        Destroy,
+        Swap,
+        Bridge
+    }
+
+    enum TransactionStatus {
+        Pending,
+        Failed,
+        Verified,
+        Finished
+    }
+
+    QtObject {
+        id: priv
```

Comment:
we use naming convention for such QtObject as "d", just to keep code consistent

Reply:
Updated.